saleor/payment/gateways/adyen/plugin.py

```diff
@@ -442,3 +479,37 @@ def capture_payment(
             error="",
             raw_response=result.message,
         )
+
+    @require_active_plugin
+    def void_payment(
+        self, payment_information: "PaymentData", previous_value
+    ) -> "GatewayResponse":
+        transaction = (
+            Transaction.objects.filter(
+                payment__id=payment_information.payment_id,
+                kind=TransactionKind.AUTH,
+                is_success=True,
+            )
+            .exclude(token__isnull=True, token__exact="")
+            .last()
+        )
+        if not transaction:
+            raise PaymentError("Cannot find a payment reference to void.")
```

Comment:
Just wondering - is this Adyen-specific or it should be part of `can_void`?

Reply:
It is specific for Adyen as any operation requires the reference received from the Auth operation.